File: specs/esm/voronoi.js
import * as vg from "@uwdata/vgplot";

await vg.coordinator().exec(
  vg.loadParquet("penguins", "data/penguins.parquet")
);

const $mesh = vg.Param.value(0);
const $hull = vg.Param.value(0);

export default vg.vconcat(
  vg.plot(
    vg.voronoi(
      vg.from("penguins"),
      {
        x: "bill_length",
        y: "bill_depth",
        stroke: "white",
        strokeWidth: 1,
        strokeOpacity: 0.5,
        inset: 1,
        fill: "species",
        fillOpacity: 0.2
      }
    ),
    vg.hull(
      vg.from("penguins"),
      {
        x: "bill_length",
        y: "bill_depth",
        stroke: "species",
        strokeOpacity: $hull,
        strokeWidth: 1.5
      }
    ),
    vg.delaunayMesh(
      vg.from("penguins"),
      {
        x: "bill_length",
        y: "bill_depth",
        z: "species",
        stroke: "species",
        strokeOpacity: $mesh,
        strokeWidth: 1
      }
    ),
    vg.dot(
      vg.from("penguins"),
      {x: "bill_length", y: "bill_depth", fill: "species", r: 2}
    ),
    vg.frame(),
    vg.inset(10),
    vg.width(680)
  ),
  vg.hconcat(
    vg.menu({
      label: "Delaunay Mesh",
      options: [{value: 0, label: "Hide"}, {value: 0.5, label: "Show"}],
      as: $mesh
    }),
    vg.hspace(5),
    vg.menu({
      label: "Convex Hull",
      options: [{value: 0, label: "Hide"}, {value: 1, label: "Show"}],
      as: $hull
    })
  )
);

File: docs/public/specs/python/voronoi.py
import vgplot as vg

penguins = vg.parquet("data/penguins.parquet")

mesh = vg.param(0)
hull = vg.param(0)

view = vg.vconcat(
    vg.plot(
        vg.voronoi(
            penguins,
            x="bill_length",
            y="bill_depth",
            stroke="white",
            stroke_width=1,
            stroke_opacity=0.5,
            fill="species",
            fill_opacity=0.2,
        ),
        vg.hull(
            penguins,
            x="bill_length",
            y="bill_depth",
            stroke="species",
            stroke_opacity=hull,
            stroke_width=1.5,
        ),
        vg.delaunay_mesh(
            penguins,
            x="bill_length",
            y="bill_depth",
            z="species",
            stroke="species",
            stroke_opacity=mesh,
            stroke_width=1,
        ),
        vg.dot(penguins, x="bill_length", y="bill_depth", fill="species", r=2),
        vg.frame(),
        vg.inset(10),
        vg.width(680),
    ),
    vg.hconcat(
        vg.menu(
            label="Delaunay Mesh",
            options=[vg.option("Hide", value=0), vg.option("Show", value=0.5)],
            bind=mesh,
        ),
        vg.hspace(5),
        vg.menu(
            label="Convex Hull",
            options=[vg.option("Hide", value=0), vg.option("Show", value=1)],
            bind=hull,
        ),
    ),
)

spec = vg.spec()
